### DISCUSSION TITLE
Syntax error, unexpected '?' on installing Presta shop basic edition 8.1.1

### DISCUSSION THREAD

**Discussion by prbt2016** [PrestaShop/PrestaShop#33879] [Q&A]
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md).
- [X] I have already [searched in existing issues](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3ABug) and found no previous report of this bug.

### Describe the bug and add attachments

Hello ,

I was in the process of manual installation of Prestashop 8.1.1 basic editon downloaded from here 
https://prestashop.com/prestashop-edition-basic/ on PHP 7.2 /7.3, MYSQL 5.7, Apache 2.4.
(Demo data enabled and all modules checked)

However I get HTTP 500 error while installation. 

So to debug further i enabled debug via config/defines.inc.php . 

So when I reran the installation the following error is thrown i.e : 

```
1/1)  ParseError syntax error, unexpected '?', expecting function (T_FUNCTION) or const (T_CONST)                                                        in ps_edition_basic.php line 29at ModuleDataProvider->PrestaShop\PrestaShop\Adapter\Module\{closure}('ps_edition_basic') in ModuleDataProvider.php line 331

```

Also following is the screenshot of the same i.e : 

![presta_err](https://github.com/PrestaShop/PrestaShop/assets/17871448/dbd23117-a9a2-4df0-895a-907bcec294ef)

Script seems to work fine with PHP 7.4. 

Also I checked the requirements here 

https://devdocs.prestashop-project.org/8/basics/installation/system-requirements/#php-requirements

Presta shop supports 7.2 and above . Are the requirements changed or is this a bug ?. 

Could you please replicate on PHP 7.2, 7.3 and fix this? . 

### Expected behavior

Presta shop should be installed without throwing HTTP 500 error 

### Steps to reproduce

1. Install presta shop 8.1.1 basic edition by entering all the required details with demo data enabled , All modules enabled.
2. HTTP 500 error thrown while module installation step. 

### PrestaShop version(s) where the bug happened

8.1.1

### PHP version(s) where the bug happened

7.2, 7.3

**Comment by Hlavtox** [comment]
Caused by `ps_edition_basic` module and low version of PHP.

**Comment by matks** [comment]
Hi, I believe this module `ps_edition_basic` needs php 7.4 or higher to work

**Comment by prbt2016** [comment]
Hello @Hlavtox  , @matks 

Thanks for the reply . 

Is this going to be fixed for compatibility with PHP 7.2, PHP 7.3 ? .

Kindly let me know.

Regards.

**Comment by joecoolio66** [comment]
Prestashop advertises compatibility with PHP 7.2 and above:

https://devdocs.prestashop-project.org/8/basics/installation/system-requirements/

**Comment by matks** [comment]
> Prestashop advertises compatibility with PHP 7.2 and above:

Prestashop 8.1.1 basic edition downloaded from here https://prestashop.com/prestashop-edition-basic/ is different from the open source project PrestaShop that you can download on GitHub 😉 . It's a specific distribution. Just like Ubuntu or Debian both are distributions of Linux.

http://prestashop-project.org/ -> the PrestaShop open source project
https://prestashop.com/ -> the PrestaShop company

For more information about PrestaShop 8.1.1 basic edition please reach out to PrestaShop company people 😉

**Comment by sallemiines** [comment]
Closing the issue as it's not bug related to Core :wink: 
@matks Could you please try to transfer this issue to discussion tab :pray:

**Comment by joecoolio66** [comment]
This is all clear as mud.  Are the prestashop basic release requirements listed anywhere then?

**Comment by matks** [comment]
> This is all clear as mud. 

Sorry 😢 I hope it will improve over time. Since both the company and the project share the same name I understand the confusion. You can better understand by reading [Beyond 1.7](https://build.prestashop-project.org/news/prestashop-beyond-1-7/) and watching the project’s [public demo from November 2021](https://www.youtube.com/watch?t=128&v=jwOy3gx-MNs&feature=youtu.be).

> Are the prestashop basic release requirements listed anywhere then?

I think it does not, but I told the team working on PrestaShop Basic Edition 😉 so hopefully they'll create one

From what I understand
- PrestaShop Basic Edition needs the module `ps_edition_basic`
- This module is not compatible with PHP7.2 and PHP7.3 - and I'm not sure if the Basic Edition team will add the compatibility with 7.2 and 7.3 or rather acknowledge this fact and document it*

*Please note PHP7.2 and PHP7.3 are [not supported](https://www.php.net/supported-versions.php) since January 2022 so using them is a bit risky
